model/src/lib.rs

```diff
@@ -170,6 +171,18 @@ impl OrderCreation {
 #[derive(Clone, Copy, Debug, Eq, Hash, PartialEq)]
 pub struct OrderUid(pub [u8; 56]);
 
+impl FromStr for OrderUid {
+    type Err = hex::FromHexError;
+    fn from_str(s: &str) -> Result<OrderUid, hex::FromHexError> {
+        let mut value = [0 as u8; 56];
+        let s_without_prefix = s
+            .strip_prefix("0x")
+            .ok_or(hex::FromHexError::InvalidStringLength)?;
```

Comment:
Here, I would like to throw a custom error stating the error more clearly.

Which error type would you guys recommend? I think I want to use a very custom error type. Should I implement my own or is there a good one to use out of the box.

Reply:
Why not allow both? `s.strip_prefix("0x").unwrap_or(s)` given that fromHex seems to work with and without the prefix?

Reply:
Yeah, this is definitively the most practical solution :) I love it.

It's just that we never did it that way and Valentin advocated for a well defined interface, and hence, we are doing:
```
let s = s.strip_prefix("0x").ok_or_else(|| {
                    de::Error::custom(format!(
                        "{:?} can't be decoded as hex uid because it does not start with '0x'",
                        s
                    ))
                 ```
                 In all the deserialize functions.

Reply:
I'm okay with allowing both too although I slightly prefer being strict. I have taken the same shortcut for allowing both for the `H160Wrapper`:
```rust
impl FromStr for H160Wrapper {
    type Err = FromHexError;
    fn from_str(s: &str) -> Result<Self, Self::Err> {
        let s = s.strip_prefix("0x").unwrap_or(s);
        Ok(H160Wrapper(H160(FromHex::from_hex(s)?)))
    }
}
```
The correct way do not allow it would be to create a new error enum which is either MissingPrefix or FromHexError probably using https://github.com/dtolnay/thiserror .